### DISCUSSION TITLE
OpenRC supported?

### DISCUSSION THREAD

**Discussion by liyimeng** [rancher/rke2#2053] [Q&A]
It seems rke2 drop openrc supported by purpose.  I am curious why. It seem being lightweight is one of the purposes of rke2, while systemd supported OS normally not, e.g. Official supported OSes in rke2, none are lightweight.

**Comment by brandond** [comment]
No, we currently only include a systemd unit. There's no reason it shouldn't work on openrc if you provide an init script though.

K3s is meant to be lightweight; RKE2 does not share this design goal. Its primary goals are to offer security, auditability, and parity with existing RKE features while building on the core K3s deployment and configuration experience.

**Comment by JustGitting** [comment]
Sorry to raise this old question, happy to start fresh if mods wish it.

>  There's no reason it shouldn't work on openrc if you provide an init script though.

Does this mean, the only thing needed is to create an openrc init.d script to start/stop/status RKE2?

> Its primary goals are to offer security, auditability, and parity ...
Are there any other gotcha's?

**Comment by termermc** [comment]
RKE2's systemd service file uses cgroup delegation, and I'm not sure there's an equivalent for that on OpenRC.

Edit: It seems like something similar does exist for OpenRC: https://wiki.gentoo.org/wiki/OpenRC/CGroups
